### ISSUE TITLE
0.1.5 breaks backwards compatibility

### ISSUE THREAD

**Issue by argaldo** [deepmipt/DeepPavlov#674]
Hi,

 I had a working goal oriented bot with 0.1.1 ... with the new release 0.1.5 it is not working anymore :(. 

This issue is for documenting what changes I had to make in my codebase to make it work with the new release, just in case the changes made on  deeppavlov/models/go_bot/network.py where not intentional.

First, the "go_bot" component in the registry is not there anymore, therefore when starting with a previous working configuration this exception is raised:

```
Traceback (most recent call last):
  File "gobot_charlotte_db.py", line 191, in <module>
    model = build_model(db_config)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/commands/infer.py", line 63, in build_model
    component = from_params(component_config, mode=mode, serialized=component_serialized)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 90, in from_params
    cls = get_model(cls_name)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/common/registry.py", line 70, in get_model
    raise ConfigError("Model {} is not registered.".format(name))
deeppavlov.core.common.errors.ConfigError: 'Model go_bot is not registered.'
```

Now the name for that component is 'go_bot_rnn' in the registry, changing the class_name to 'go_bot_rnn' in the configuration is enough, like so:

**before:** 

```
bot_comp_config = {
    'class_name': 'go_bot',
```

**after:**

```
bot_comp_config = {
    'class_name': 'go_bot_rnn',
```

Then, when the component is found and the bot is starting one gets this exception:

```
2018-12-28 12:03:27.507 ERROR in 'deeppavlov.core.common.params'['params'] at line 106: Exception in <class 'deeppavlov.models.go_bot.network.GoalOrientedBotNetwork'>
Traceback (most recent call last):
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 100, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/models/tf_backend.py", line 77, in __call__
    obj.__init__(*args, **kwargs)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/models/tf_backend.py", line 30, in _wrapped
    return func(*args, **kwargs)
TypeError: __init__() missing 1 required positional argument: 'hidden_size'
Traceback (most recent call last):
  File "gobot_charlotte_db.py", line 191, in <module>
    model = build_model(db_config)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/commands/infer.py", line 63, in build_model
    component = from_params(component_config, mode=mode, serialized=component_serialized)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 100, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/models/tf_backend.py", line 77, in __call__
    obj.__init__(*args, **kwargs)
  File "/home/argaldo/deeppavlov/env/lib/python3.6/site-packages/deeppavlov/core/models/tf_backend.py", line 30, in _wrapped
    return func(*args, **kwargs)
TypeError: __init__() missing 1 required positional argument: 'hidden_size'
```

You can trace back this error to a problem extracting keyword parameters in GoalOrientedBotNetwork constructor, it is looking for "hidden_size", "dense_size",.. among others. The problem is that in the previous version this was a dictionary with all the parameters of the component and "dense_size", "hidden_size", "attention_mechanism" ... where included as a dictionary named "network_parameters" in the configuration dictionary:

**before ( snippet ):**

```
[...]
    'database': None,
    'api_call_action': 'api_call',
    'network_parameters': {
      'dense_size': 64,
      'hidden_size': 128,
      'learning_rate': 0.002,
      'attention_mechanism': None
    },
    'tokenizer': {
[...]
```
I had to modify that dictionary to pass the network parameters not in a dictionary, but as keyword parameters, like so:

**after ( snippet, note the commented lines and extra comma ):**

```
[...]
    'database': None,
    'api_call_action': 'api_call',
    #'network_parameters': {
      'dense_size': 64,
      'hidden_size': 128,
      'learning_rate': 0.002,
      'attention_mechanism': None,
    #},
    'tokenizer': {
[...]
```

After making this changes, the bot finally starts and works as before.

In case this wasn't intentional and is a bug, making the codebase backwards compatible at least for the keyword parameters for GoalOrientedBotNetwork is easy and has been submitted as a pull request for your consideration, here-> [#673]( https://github.com/deepmipt/DeepPavlov/pull/673 )

**Comment by vikmary** [comment]
https://github.com/deepmipt/DeepPavlov/pull/675
Please check if your old configs (without any changing) are working in the pull-request.

**Comment by argaldo** [comment]
I tried with the "fix/go-bot-0.1.0-compatibility" branch and it indeed works now with my previous bot configuration, both issues, "go_bot" component registry entry and named parameters seem to be resolved. Thank you very much.

**Comment by seliverstov** [comment]
Fixed in 1.5.1
